opentelemetry-auto-instrumentation/src/opentelemetry/auto_instrumentation/instrumentor.py

```diff
@@ -38,14 +48,35 @@ def __new__(cls):
 
     @abstractmethod
     def _instrument(self, **kwargs):
-        """Instrument"""
+        """Instrument the library"""
 
     @abstractmethod
     def _uninstrument(self, **kwargs):
-        """Uninstrument"""
+        """Uninstrument the library"""
 
     def instrument(self, **kwargs):
-        """Instrument"""
+        """Instrument the library
+
+        This method will be called without any optional arguments by the
+        ``opentelemetry-auto-instrumentation`` command. The configuration of
+        the instrumentation when done in this way should be done by previously
+        setting the configuration (using environment variables or any other
+        mechanism) that will be used later by the code in the ``instrument``
+        implementation via the global ``Configuration`` object.
+
+        When this method is to be called directly in the user code, ``kwargs``
```

Comment:
I feel like the following three paragraphs could be a bit more succint:

```
The instrument method kwargs should default to values from the Configuration object. However, this method should not modify the Configuration object in any way.
```

I feel like it's clear from context that opentelemetry-auto-instrumentation command line behavior is identical to instrument() with no arguments.

Reply:
Updated the documentation with your comments. Nevertheless, I prefer to keep the paragraph that explains that using ``instrument`` with no arguments should do the very same thing that calling it from ``opentelemetry-auto-instrumentation``, I find this an example that even when obvious to you, it may not be so for others and actually helps understand the intention behind the design of the ``BaseInstrumentor`` class.